src/overlay.js

```diff
@@ -105,29 +111,35 @@
             options.location.x,
             options.location.y,
             options.location.width,
-            options.location.height
-        );
-        this.position   = new $.Point(
-            options.location.x,
-            options.location.y
-        );
-        this.size       = new $.Point(
-            options.location.width,
-            options.location.height
-        );
-        this.style      = options.element.style;
-        // rects are always top-left
-        this.placement  = options.location instanceof $.Point ?
+            options.location.height);
+
+        // this.position is never read by this class but is kept for backward
+        // compatibility
+        this.position = this.bounds.getTopLeft();
+
+        // this.size is only used by PointOverlay with options.checkResize === false
+        this.size = this.bounds.getSize();
+
+        this.style = options.element.style;
+
+        // rects are always top-left (RectOverlays don't use placement)
```

Comment:
If they don't scale in one dimension, should they support placement in that dimension?